index.js

```diff
@@ -22,6 +39,18 @@ export default function pTimeout(promise, milliseconds, fallback, options) {
 			...options
 		};
 
+		if (options && options.signal) {
+			const {signal} = options;
+
+			signal.addEventListener('abort', () => {
```

Comment:
Can you add a TODO to use https://developer.mozilla.org/en-US/docs/Web/API/AbortSignal/throwIfAborted when targeting Node.js 18?

Reply:
I think probably you missed [this comment](https://github.com/sindresorhus/got/pull/2020#discussion_r846141810)